mods/default/nodes.lua

```diff
@@ -586,29 +586,107 @@ minetest.register_node("default:snow", {
 	inventory_image = "default_snowball.png",
 	wield_image = "default_snowball.png",
 	paramtype = "light",
+	paramtype2 = "leveled",
 	buildable_to = true,
 	floodable = true,
+	walkable = true,
 	drawtype = "nodebox",
+	leveled = 16,
 	node_box = {
-		type = "fixed",
-		fixed = {
-			{-0.5, -0.5, -0.5, 0.5, -0.25, 0.5},
-		},
+		type = "leveled",
+		fixed = {-0.5, -0.5, -0.5, 0.5, -0.5, 0.5},
 	},
 	collision_box = {
-		type = "fixed",
-		fixed = {
-			{-0.5, -0.5, -0.5, 0.5, -7 / 16, 0.5},
-		},
+		type = "leveled",
+		fixed = {-0.5, -0.5, -0.5, 0.5, -0.5, 0.5},
+	},
+	selection_box = {
+		type = "leveled",
+		fixed = {-0.5, -0.5, -0.5, 0.5, -0.5, 0.5},
 	},
 	groups = {crumbly = 3, falling_node = 1, snowy = 1},
 	sounds = default.node_sound_snow_defaults(),
+	node_placement_prediction = nil,
 
 	on_construct = function(pos)
```

Comment:
Testing shows, that this is called when falling snow stacks.
This is possible:
```lua
on_construct = function(pos)
	local level = minetest.get_node_level(pos)
	if level > 16 then
		minetest.sound_play(default.node_sound_snow_defaults().place,
				{pos = pos, max_hear_distance = 8})
	end
	if level >= 64 then
		minetest.set_node(pos, {name="default:snowblock"})
	end
```
This even adds sounds for `node_placement_prediction = ""` (Then the `if level > 16 then` should be removed.).

Reply:
So replace on_construct with that code?

Reply:
No, just _add_ code:
```lua
on_construct = function(pos)
	local level = minetest.get_node_level(pos)
	if level >= 64 then
		minetest.set_node(pos, {name="default:snowblock"})
	end

	pos.y = pos.y - 1
...
```

The placement sound shouldn't always be played at a position. I'll write about that in a separate comment.

Reply:
Done!